types/react-dom/experimental.d.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * These are types for things that are present in the `experimental` builds of React but not yet
+ * on a stable build.
+ *
+ * Once they are promoted to stable they can just be moved to the main index file.
+ *
+ * To load the types declared here in an actual project, there are three ways. The easiest one,
+ * if your `tsconfig.json` already has a `"types"` array in the `"compilerOptions"` section,
+ * is to add `"react-dom/experimental"` to the `"types"` array.
+ *
+ * Alternatively, a specific import syntax can to be used from a typescript file.
+ * This module does not exist in reality, which is why the {} is important:
+ *
+ * ```ts
+ * import {} from 'react-dom/experimental'
+ * ```
+ *
+ * It is also possible to include it through a triple-slash reference:
+ *
+ * ```ts
+ * /// <reference types="react-dom/experimental" />
+ * ```
+ *
+ * Either the import or the reference only needs to appear once, anywhere in the project.
+ */
+
+// See https://github.com/facebook/react/blob/master/packages/react-dom/src/client/ReactDOM.js to see how the exports are declared,
+// and https://github.com/facebook/react/blob/master/packages/shared/ReactFeatureFlags.js to verify which APIs are
+// flagged experimental or not. Experimental APIs will be tagged with `__EXPERIMENTAL__`.
+
+import React = require('react');
+import ReactDOM = require('.');
+
+export {};
+
+declare module '.' {
+    interface Work {
+        then(onCommit?: () => void): void;
+    }
+
+    interface Batch {
+        commit(): void;
+        render(children: React.ReactChild | React.ReactNodeArray): Work;
+        then(onComplete?: () => void): void;
+    }
+
+    interface RootOptions {
+        hydrate?: boolean;
+    }
+
+    interface BlockingRoot {
```

Comment:
As far as I know the interface for the roots returned from createRoot and createBlockingRoot are identical. We should remove the createRoot typings from the index.d.ts anyway since they got [removed in 16.10](https://github.com/facebook/react/blob/master/CHANGELOG.md#react-dom) and then  we can reuse them here.